### ISSUE TITLE
Autogenerated help message through `az find "az aks enable-addons"` is wrong

### ISSUE THREAD

**Issue by MikkelHegn** [Azure/azure-cli#15442]
## Describe the bug

**Command Name**
`az find "az aks enable-addons"`

**Errors:**
The autogenerated example is wrong:

```
Enable Kubernetes addons. (autogenerated)
az aks enable-addons --addons virtual-node --name MyManagedCluster --resource-group MyResourceGroup
```

## To Reproduce:
Steps to reproduce the behavior.

- `az find "az aks enable-addons"`

## Expected Behavior
To show the example from the `az aks enable-addons --help` command. Subnet as a required argument is missing.

```
Examples
    Enable Kubernetes addons. (autogenerated)
        az aks enable-addons --addons virtual-node --name MyManagedCluster --resource-group
        MyResourceGroup --subnet MySubnetName
```

## Environment Summary
```
Linux-4.19.128-microsoft-standard-x86_64-with-debian-buster-sid
Python 3.6.10
Installer: DEB

azure-cli 2.12.1

```
## Additional Context

The message from `az aks enable-addons --help` was addressed in #12871

**Comment by yungezz** [comment]
hi @mirdaki @christopher-o-toole could you pls have a look at this issue?

**Comment by mirdaki** [comment]
I've roped in the relevant people. Will update when I have something.

**Comment by MikkelHegn** [comment]
Any updates? This is still a problem with 2.15.1

**Comment by mirdaki** [comment]
@MikkelHegn so sorry for the delay. We were trying to figure out if we could solve this sort of edge case if a general way, which took more time than anticipated. The updates should be live now (caches might not be updated everywhere).

**Comment by MikkelHegn** [comment]
Thx - I see this is fixed in 2.16.0
